wpilib/wpilib/command/scheduler.py

```diff
@@ -217,52 +217,31 @@ def enable(self):
         """
         self.disabled = False
 
-    def getName(self):
-        return "Scheduler"
-
-    def getType(self):
-        return "Scheduler"
-
-    def initTable(self, subtable):
-        self.commands = []
-        self.ids = []
-        self.toCancel = []
-
-        if subtable is not None:
-            self.namesEntry = subtable.getEntry("Names")
-            self.idsEntry = subtable.getEntry("Ids")
-            self.cancelEntry = subtable.getEntry("Cancel")
-
-            self.namesEntry.setStringArray(self.commands)
-            self.idsEntry.setDoubleArray(self.ids)
-            self.cancelEntry.setDoubleArray(self.toCancel)
-        else:
-            self.namesEntry = None
-            self.idsEntry = None
-            self.cancelEntry = None
+    def initSendable(self, builder):
+        builder.setSmartDashboardType("Scheduler")
+        self.namesEntry = builder.getEntry("Names")
+        self.idsEntry = builder.getEntry("Ids")
+        self.cancelEntry = builder.getEntry("Cancel")
+        builder.setUpdateTable(self.updateTable)
 
     def updateTable(self):
-        if not (self.namesEntry is not None and self.cancelEntry is not None):
+        if not (self.namesEntry is not None and self.idsEntry is not None and self.cancelEntry is not None):
             return
 
-        # Get the commands to cancel
-        self.toCancel = self.cancelEntry.getDoubleArray([])
-        if self.toCancel:
+        toCancel = self.cancelEntry.getDoubleArray([])
+        if toCancel:
             for command in self.commandTable:
-                if id(command) in self.toCancel:
+                if float(id(command)) in toCancel:
```

Comment:
```python
>>> 1 in [1.0]
True
```

Reply:
-9223363293403863878 in [float(-9223363293403863878)]

Reply:
```python
>>> -9223363293403863878 .bit_length()
-63
```

So that's not surprising. I don't think the heap can go that far.

Reply:
I assume the cause is at that value the distance between consecutive numbers on the ieee754 number line is greater than 1. Don't know (or care) what it is, or where the cutoff point is, but the float conversion prevents it from being an issue

Reply:
The fact that this will never happen in reality also prevents it from being an issue.